metadata/moa/scripts/nbconverted/2.map-broad_id.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import rdkit.Chem.inchi as inchi
+
+
+# Sample files from each version of the repurposing hub are read
+
+# In[2]:
+
+
+samples_2017 = pd.read_csv('clue/repurposing_samples_20170327.txt',usecols=['broad_id','pert_iname','InChIKey'],delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2018a = pd.read_csv('clue/repurposing_samples_20180516.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2018b = pd.read_csv('clue/repurposing_samples_20180907.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2020 = pd.read_csv('clue/repurposing_samples_20200324.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+
+
+# In[3]:
+
+
+# 2017 version is missing deprecated_broad_id
+samples_2017['deprecated_broad_id'] = np.nan
+
+
+# In[4]:
+
+
+# The first 13 characters of the Broad ID and the first 14 characters of InChIKey are extracted
+# Year names are appended to column names
+# For some rows InChI are extracted instead of InChIKey -> these are fixed
+
+def id_cleanup(df, year):
+    df.dropna(subset=['InChIKey'], inplace=True)
+    df.reset_index(drop=True, inplace=True)
+    df.InChIKey = df.InChIKey.apply(lambda x: inchi.InchiToInchiKey(x) if (x.startswith('InChI')) else x)
+    df.broad_id = df.broad_id.apply(lambda x: str(x)[:13])
+    df.InChIKey = df.InChIKey.apply(lambda x: str(x)[:14])
+    df = df.drop_duplicates(['InChIKey','pert_iname','broad_id','deprecated_broad_id']).reset_index(drop=True)
+    df = df.rename(columns={'pert_iname':'pert_iname_'+year,
```

Comment:
up front note - in looking this up, I actually discovered that my [current method of string formatting](https://github.com/broadinstitute/lincs-cell-painting/blob/master/metadata/moa/scripts/nbconverted/0.merge-repurposing-compounds.py#L31) is also suboptimal! (This is one super useful benefit of code review)

Check out updated methods to format strings [here](https://realpython.com/python-f-strings/). 

so it looks like `'pert_iname_'+year` becomes `f"pert_iname_{year}"`.

Reply:
This is great! I was just now slowly moving towards str.format() but the updated method is pretty sweet.